tests/phpunit/tests/blocks/setIgnoredHookedBlocksMetadata.php

```diff
@@ -144,4 +144,36 @@ public function test_set_ignored_hooked_blocks_metadata_for_block_added_by_conte
 			$anchor_block['attrs']['metadata']['ignoredHookedBlocks']
 		);
 	}
+
+	/**
+	 * @ticket 60580
+	 *
+	 * @covers ::set_ignored_hooked_blocks_metadata
+	 */
+	public function test_set_ignored_hooked_blocks_metadata_for_block_suppressed_by_filter() {
```

Comment:
I had a hard time understanding why the `tests/hooked-block-suppressed-by-filter` block type got removed from the list of ignored hooked blocks. I realized it's related to the filter name, so maybe it should be made more obvious:

```suggestion
	public function test_set_ignored_hooked_blocks_metadata_for_specific_block_type_suppressed_by_filter() {
```

Another idea would be to put the name of the block in a variable and using it in the expression, example:

```php
add_filter( 'hooked_block_' + $block_suppressed_by_filter, $filter, 10, 4 );
```

Reply:
Ah, good point. I had an idea for a slightly different test that should be a bit easier to follow; I'll give that a try.

Reply:
It can be a follow-up after Beta 2 :)

Reply:
Here's a minor rewrite that should make things a bit easier to read:

```diff
diff --git a/tests/phpunit/tests/blocks/setIgnoredHookedBlocksMetadata.php b/tests/phpunit/tests/blocks/setIgnoredHookedBlocksMetadata.php
index d5e9c4e23f..7145e56cde 100644
--- a/tests/phpunit/tests/blocks/setIgnoredHookedBlocksMetadata.php
+++ b/tests/phpunit/tests/blocks/setIgnoredHookedBlocksMetadata.php
@@ -163,16 +163,20 @@ class Tests_Blocks_SetIgnoredHookedBlocksMetadata extends WP_UnitTestCase {
                );
 
                $filter = function ( $parsed_hooked_block, $hooked_block_type, $relative_position, $parsed_anchor_block ) {
-                       if ( 'tests/anchor-block' === $parsed_anchor_block['blockName'] && 'after' === $relative_position ) {
+                       if (
+                               'tests/hooked-block-suppressed-by-filter' === $hooked_block_type &&
+                               'after' === $relative_position &&
+                               'tests/anchor-block' === $parsed_anchor_block['blockName']
+                       ) {
                                return null;
                        }
 
                        return $parsed_hooked_block;
                };
 
-               add_filter( 'hooked_block_tests/hooked-block-suppressed-by-filter', $filter, 10, 4 );
+               add_filter( 'hooked_block', $filter, 10, 4 );
                set_ignored_hooked_blocks_metadata( $anchor_block, 'after', $hooked_blocks, null );
-               remove_filter( 'hooked_block_tests/hooked-block-suppressed-by-filter', $filter );
+               remove_filter( 'hooked_block', $filter );
 
                $this->assertSame( array( 'tests/hooked-block' ), $anchor_block['attrs']['metadata']['ignoredHookedBlocks'] );
        }
```

Reply:
That would help, indeed 💯

Reply:
Committed to Core in https://core.trac.wordpress.org/changeset/57676.